FirebaseSegmentation/Sources/Private/FIRSegmentationComponent.h

```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright 2019 Google
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#import <Foundation/Foundation.h>
+
+#import "FIRLibrary.h"
+
+@class FIRApp;
+@class FIRSegmentation;
+
+NS_ASSUME_NONNULL_BEGIN
+
+/// Provides and creates instances of Segmentation. Used in the
+/// interop registration process to keep track of Segmentation instances for each `FIRApp` instance.
+@protocol FIRSegmentationProvider
+
+/// Cached instances of Segmentation objects.
+@property(nonatomic, strong) NSMutableDictionary<NSString *, FIRSegmentation *> *instances;
```

Comment:
I would add an additional method like
`- (void)insertSegmentation:(FIRSegmentation *)segmentation withName:(NSString *)name`
so we can get rid of mutable dictionary and make it as simple **NSDictionary** which will make this property more consistency and more encapsulated.
In 
`- (void)insertSegmentation:(FIRSegmentation *)segmentation withName:(NSString *)name` would be something like:
```
synchronized(self) {
 if (name) {
self.mutableInstances[name] = segmentation;
}
}
```